### PR TITLE
feat: sketching Cancel Stuck Backups endpoint

### PR DESCRIPTION
**Pull request by moogacs**
### What's being changed:
this PR sketch the cancel backup in progress with designing the REST endpoint params and response
this endpoint is needed in case we have stuck backups we can cancel them

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: openapi-specs/schema.json [hunk 1]
```diff
@@ -4858,6 +4858,58 @@
...
+            "required": true,
+            "type": "string",
+            "description": "Backup backend name e.g. filesystem, gcs, s3."
+          },
+          {
+            "name": "id",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The ID of a backup. Must be URL-safe and work as a filesystem path, only lowercase, numbers, underscore, minus characters allowed."
```
Comment (reyreaud-l) [inline openapi-specs/schema.json:4884]: Is there any place in the code where validate the backup ID ? I'm assuming we do it when we create a backup. Will it be a problem here if we use whichever user input ?

File: usecases/backup/scheduler.go [hunk 2]
```diff
@@ -215,6 +215,27 @@ func (s *Scheduler) RestorationStatus(ctx context.Context, principal *models.Pri
...
 }
 
+func (s *Scheduler) Cancel(ctx context.Context, principal *models.Principal, backend, backupID string,
+) error {
+	defer func(begin time.Time) {
+		var err error
+		logOperation(s.logger, "cancel_backup", backupID, backend, begin, err)
+	}(time.Now())
+
+	path := fmt.Sprintf("backups/%s/%s", backend, backupID)
```
Comment (reyreaud-l) [inline usecases/backup/scheduler.go:225]: backupID is not sanitized right ? Maybe with the right input we could have a vulnerability here when building the string.

Do you think it' a concern or something we'll tackle later ?

**Review by reyreaud-l — APPROVED** [review]
LGTM, see comments i'm wondering about sanytizing user input to avoid malicious usage

File: usecases/backup/scheduler.go (hunk 2 above)
Comment (moogacs) [reply usecases/backup/scheduler.go:225]: yeah i agree, will take care of it when have the real cancelation  impl

Comment (moogacs) [reply usecases/backup/scheduler.go:225]: however i think it's just the url safety but it won't be a security concern

Comment (moogacs) [reply usecases/backup/scheduler.go:225]: will take care of it in a follow up PR

**Pull request merged**